### PR TITLE
Add binarized_mnist to TFDS.

### PR DESCRIPTION
**Pull request by tfds-copybara**
Add binarized_mnist to TFDS.

### CODE REVIEW

**Comment by us** [comment]
Related issue #758.

File: tensorflow_datasets/image/binarized_mnist.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  VERSION = tfds.core.Version("1.0.0")
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "image": tfds.features.Image(
+                shape=mnist.MNIST_IMAGE_SHAPE)}),
+        urls=[self.URL],
```
Comment (us) [inline tensorflow_datasets/image/binarized_mnist.py]: This url should be dataset page link, not download link.

**Comment by us** [comment]
@gjtucker why didn't inherit anything from `Mnist` class?

File: tensorflow_datasets/core/download/checksums.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def _get_sizes_checksums(checksums_path):
   """Returns {URL: (size, checksum)}s stored within file."""
   checksums = {}
   for line in _read_file(checksums_path).split('\n'):
-    if not line:
+    if not line or line.strip().startswith('#'):
```
Comment (us) [inline tensorflow_datasets/core/download/checksums.py]: Commits must do a particular job. you have already made this change on #808.

Comment (gjtucker) [reply tensorflow_datasets/core/download/checksums.py]: Fixed

**Comment by gjtucker** [comment]
> @gjtucker why didn't inherit anything from `Mnist` class?

I started that way, but ended up using none of the super class so changed it not to inherit.

File: tensorflow_datasets/image/binarized_mnist.py (hunk 1 above)
Comment (gjtucker) [reply tensorflow_datasets/image/binarized_mnist.py]: Unfortunately, I don't know of one.

**Pull request merged**